Quadriga/src/main/java/edu/asu/spring/quadriga/service/publicwebsite/impl/ProjectStats.java

```diff
@@ -0,0 +1,198 @@
+                                node.getDescription(), node.getLabel(), 1);
+                        mapStats.put(url, conceptStats);
+                    }
+                }
+            }
+        }
+
+        List<IConceptStats> valuesList = new ArrayList<IConceptStats>(
+                mapStats.values());
+
+        return getSortedList(valuesList);
+
+    }
+
+    public List<IConceptStats> getSortedList(List<IConceptStats> csList) {
+        Collections.sort(csList, new Comparator<IConceptStats>() {
+            public int compare(IConceptStats o1, IConceptStats o2) {
+                if (o1.getCount() == o2.getCount())
+                    return 0;
+                return o1.getCount() < o2.getCount() ? 1 : -1;
+            }
+        });
+        return csList;
+    }
+
+    @Override
+    public Map<String, IUserStats> getCountofWorkspace(String projectId)
+            throws QuadrigaStorageException {
+
+        List<IProjectCollaborator> projectCollaboratorList = projectCollaboratorManager
+                .getProjectCollaborators(projectId);
+
+        if (projectCollaboratorList == null)
+            return null;
+
+        Map<String, IUserStats> mapUserWorkspace = new HashMap<String, IUserStats>();
+        // loop through each collaborator
+
+        for (IProjectCollaborator projectCollaborator : projectCollaboratorList) {
+            if (projectCollaborator.getCollaborator() != null) {
+                String username = projectCollaborator.getCollaborator()
+                        .getUserObj().getUserName();
+                UserStats userStats = (UserStats) setUserWorkspaceStats(
+                        projectId, username);
+                mapUserWorkspace.put(username, userStats);
+            }
+        }
+
+        String username = projectDeepMapper.getProjectDetails(projectId)
+                .getOwner().getUserName();
+        if (username == null)
+            return mapUserWorkspace;
+
+        UserStats userStats = (UserStats) setUserWorkspaceStats(projectId,
+                username);
+        mapUserWorkspace.put(username, userStats);
+
+        return mapUserWorkspace;
+    }
+
+    public IUserStats setUserWorkspaceStats(String projectId, String username)
+            throws QuadrigaStorageException {
+        List<IWorkSpace> workspaceList = listWSDAO.listWorkspace(projectId,
+                username);
+        UserStats userStats;
+        if (workspaceList == null) {
+            userStats = new UserStats(username, 0, 0);
+        } else {
+            userStats = new UserStats(username, workspaceList.size(), 0);
+        }
+        return userStats;
+
+    }
+
+    @Override
+    public Map<String, IUserStats> getCountofNetwork(String projectid)
+            throws QuadrigaStorageException {
+        List<INetwork> networksList = networkManager
+                .getNetworksInProject(projectid);
+
+        if (networksList == null)
+            return null;

```

Comment:
I would return an empty list here, since there are effectively no network statistics available for the project.